swagger_templates/python/api.mustache

```diff
@@ -0,0 +1,221 @@
+        """
+        kwargs['_return_http_data_only'] = True
+        if kwargs.get('callback'):
+            return self.{{operationId}}_with_http_info({{#sortParamsByRequiredFlag}}{{#allParams}}{{#required}}{{paramName}}, {{/required}}{{/allParams}}{{/sortParamsByRequiredFlag}}**kwargs)
+        else:
+            (data) = self.{{operationId}}_with_http_info({{#sortParamsByRequiredFlag}}{{#allParams}}{{#required}}{{paramName}}, {{/required}}{{/allParams}}{{/sortParamsByRequiredFlag}}**kwargs)
+            return data
+
+    def {{operationId}}_with_http_info(self, {{#sortParamsByRequiredFlag}}{{#allParams}}{{#required}}{{paramName}}, {{/required}}{{/allParams}}{{/sortParamsByRequiredFlag}}**kwargs):
+        """
+        {{{summary}}}
+        {{{notes}}}
+
+        This method makes a synchronous HTTP request by default. To make an
+        asynchronous HTTP request, please define a `callback` function
+        to be invoked when receiving the response.
+        >>> def callback_function(response):
+        >>>     pprint(response)
+        >>>
+{{#sortParamsByRequiredFlag}}
+        >>> thread = api.{{operationId}}_with_http_info({{#allParams}}{{#required}}{{paramName}}, {{/required}}{{/allParams}}callback=callback_function)
+{{/sortParamsByRequiredFlag}}
+{{^sortParamsByRequiredFlag}}
+        >>> thread = api.{{operationId}}_with_http_info({{#allParams}}{{#required}}{{paramName}}={{paramName}}_value, {{/required}}{{/allParams}}callback=callback_function)
+{{/sortParamsByRequiredFlag}}
+
+        :param callback function: The callback function
+            for asynchronous request. (optional)
+{{#allParams}}
+        :param {{dataType}} {{paramName}}: {{{description}}}{{#required}} (required){{/required}}{{#optional}}(optional{{#defaultValue}}, default to {{{.}}}{{/defaultValue}}){{/optional}}
+{{/allParams}}
+        :return: {{#returnType}}{{returnType}}{{/returnType}}{{^returnType}}None{{/returnType}}
+                 If the method is called asynchronously,
+                 returns the request thread.
+        """
+
+        all_params = [{{#allParams}}'{{paramName}}'{{#hasMore}}, {{/hasMore}}{{/allParams}}]
+        all_params.append('callback')
+        all_params.append('_return_http_data_only')
+
+        params = locals()
+        for key, val in iteritems(params['kwargs']):
+            if key not in all_params:
+                raise TypeError(
+                    "Got an unexpected keyword argument '%s'"
+                    " to method {{operationId}}" % key
+                )
+            params[key] = val
+        del params['kwargs']
+{{#allParams}}
+{{#required}}
+        # verify the required parameter '{{paramName}}' is set
+        if ('{{paramName}}' not in params) or (params['{{paramName}}'] is None):
+            raise ValueError("Missing the required parameter `{{paramName}}` when calling `{{operationId}}`")
+{{/required}}
+{{/allParams}}
+
+{{#allParams}}
+{{#vendorExtensions.x-isi-url-encode-path-param}}
```

Comment:
FYI, this is where I stuck the code that gets generated when x-isi-url-encode-path-param.